spec.html

```diff
@@ -51929,6 +51994,8 @@ <h1>Regular Expressions</h1>
 <emu-annex id="sec-additional-ecmascript-features-for-web-browsers" namespace="annexB" normative>
   <h1>Additional ECMAScript Features for Web Browsers</h1>
   <p>The ECMAScript language syntax and semantics defined in this annex are required when the ECMAScript host is a web browser. The content of this annex is normative but optional if the ECMAScript host is not a web browser.</p>
+  <p>Some features defined in this annex are specified in this annex, and some are specified in the main body of this document.</p>
+  <p>When a feature is specified in the main body, each point where it affects the document is marked with the words "Normative Optional" in a coloured box. Moreover, where the feature involves particular wording in an algorithm or early error rule, this is guarded by the condition that “<dfn variants="otherwise supports">the host supports</dfn>” the relevant feature. Web browsers are required to support all such features.</p>
```

Comment:
Is it the case that web browsers are required to support all normative optional features? (I think yes?) If so, we should say so in the definition of "normative optional", not here.

Reply:
That may be true today, but I don't know if we want it to always be true. If it's ever not true, we run the risk of forgetting to update that text. I think the mention here is fine.

Reply:
I would prefer to optimize for clarity of the existing document over guarding against future editorial mistakes.

Reply:
I think it is conceptually more appropriate here as well. I don't want anyone learning "normative optional" means "available in web browsers" when that may not be true in the future, even if it is editorially corrected.

Reply:
My concern is that I don't want people to have to find this sentence in the annex in order to know that these features are in fact required. The fact that it's currently easy to miss that Annex B imposes additional requirements on web browsers (and therefore any engine which wants to run code written for web browsers) is the whole reason we're doing this in the first place.

I really don't think we should be worrying about being defensive against a future world where we have additional normative optional features which are optional for browsers.

Reply:
Per editor call, we can add a sentence at the definition of "normative optional" along the lines of "Web browsers are generally required to implement all normative optional features."

Reply:
Done.

I changed "features" to "subclauses" to fit better with the wording in the definition.

And for the reader dissatisfied with "generally", I added "(See Annex B.)"